src/Builder/ClassModifier/ClassInstanceModifierStrategy/ClassInstanceModifierStrategyInterface.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Integrator\Builder\ClassModifier\ClassInstanceModifierStrategy;
+
+use PhpParser\Node\Stmt\ClassMethod;
+use SprykerSdk\Integrator\Transfer\ClassInformationTransfer;
+use SprykerSdk\Integrator\Transfer\ClassMetadataTransfer;
+
+interface ClassInstanceModifierStrategyInterface
+{
+    /**
+     * @param \PhpParser\Node\Stmt\ClassMethod $node
+     *
+     * @return bool
+     */
+    public function isApplicable(ClassMethod $node): bool;
+
+    /**
+     * @param \SprykerSdk\Integrator\Transfer\ClassInformationTransfer $classInformationTransfer
+     * @param string $targetMethodName
+     * @param \SprykerSdk\Integrator\Transfer\ClassMetadataTransfer $classMetadataTransfer
+     *
+     * @return \SprykerSdk\Integrator\Transfer\ClassInformationTransfer
+     */
+    public function wireClassInstance(
```

Comment:
What do you think about strategy responsibilities?

Reply:
To my mind, we have to follow "single responsibilities" principle and follow rules that are already applied in the project. 
We have `ClassInstanceClassModifierInterface` thats contains `wireClassInstance` and `unwireClassInstance` methods. 
To my mind, here better to follow the structure as it is. 
https://github.com/spryker-sdk/integrator/blob/feature/PPLUS-1141-plugin-collections/src/Builder/ClassModifier/ClassInstanceClassModifierInterface.php

Reply:
And what do you think, does it violate SRP with wire and unwire methods?
Does it violate any other principles?

Reply:
These interfaces were divided, please see https://github.com/spryker-sdk/integrator/pull/42/files